integration/querydsl/blaze-persistence-querydsl-testsuite/src/test/java/com/blazebit/persistence/querydsl/BasicQueryTest.java

```diff
@@ -0,0 +1,744 @@
+            List<Tuple> fetch = queryable.getResultList();
+            assertFalse(fetch.isEmpty());
+        });
+    }
+
+    @Test
+    public void testThroughBlazeJPAQuery() {
+        doInJPA(entityManager -> {
+            BlazeJPAQuery<Tuple> query = new BlazeJPAQuery<Tuple>(entityManager, cbf).from(document)
+                    .select(document.name.as("blep"), document.name.substring(0, 2))
+                    .where(document.name.length().gt(1));
+
+            List<Tuple> fetch = query.fetch();
+            assertFalse(fetch.isEmpty());
+        });
+    }
+
+    @Test
+    public void testParameterExpression() {
+        doInJPA(entityManager -> {
+            Param<Integer> param = new Param<>(Integer.class, "theSuperName");
+
+            BlazeJPAQuery<Tuple> query = new BlazeJPAQuery<Document>(entityManager, cbf).from(document)
+                    .select(document.name.as("blep"), document.name.substring(0, 2))
+                    .where(document.name.length().gt(param))
+                    .set(param, 1);
+
+            List<Tuple> fetch = query.fetch();
+            assertFalse(fetch.isEmpty());
+        });
+    }
+
+    @Test
+    public void testParameterExpressionInSelect() {
+        doInJPA(entityManager -> {
+            Param<Integer> param = new Param<>(Integer.class, "theSuperName");
+
+            BlazeJPAQuery<Tuple> query = new BlazeJPAQuery<Document>(entityManager, cbf).from(document)
+                    .select(document.name.as("blep"), param)
+                    .where(document.name.length().gt(param))
+                    .set(param, 1);
+
+            List<Tuple> fetch = query.fetch();
+            assertFalse(fetch.isEmpty());
+        });
+    }
+
+    @Test
+    public void testSubQuery() {
+        doInJPA(entityManager -> {
+            QDocument sub = new QDocument("sub");
+            BlazeJPAQuery<Tuple> query = new BlazeJPAQuery<Document>(entityManager, cbf).from(document)
+                    .select(document.name.as("blep"), document.name.substring(0, 2))
+                    .where(document.id.in(select(sub.id).from(sub)));
+
+            List<Tuple> fetch = query.fetch();
+            assertFalse(fetch.isEmpty());
+        });
+    }
+
+    @Test
+    @Category({ NoMySQLOld.class })
+    public void testWindowFunction() {
+        Assume.assumeTrue(dbmsDialect.supportsWindowFunctions());
+
+        doInJPA(entityManager -> {
+            QDocument sub = new QDocument("sub");
+
+            BlazeJPAQuery<Tuple> query = new BlazeJPAQuery<Document>(entityManager, cbf).from(document)
+                    .select(document.name.as("documentName"), rowNumber().over().orderBy(document.id), lastValue(document.name).over().partitionBy(document.id).orderBy(document.id))
+                    .where(document.id.in(select(sub.id).from(sub)));
+
+            List<Tuple> fetch = query.fetch();
+            assertFalse(fetch.isEmpty());
+        });
+    }
+
+    @Test
+    @Category({ NoMySQLOld.class })
```

Comment:
Again, add a comment like e.g. `// NOTE: Window functions were only introduced in MySQL8`